cocos/tiledmap/tiled-layer.ts

```diff
@@ -53,6 +53,10 @@ const _vec3_temp = new Vec3();
 const _vec3_temp2 = new Vec3();
 const _tempRowCol = { row: 0, col: 0 };
 
+function isTiledSubNodeData (data: TiledRenderData | TiledSubNodeData): data is TiledSubNodeData {
```

Comment:
`data is TiledSubNodeData`: What does this code mean?

Reply:
It's similar with the PR https://github.com/cocos/cocos-engine/pull/18264
Basically, this is a typescript syntax that if the function returns `true`, then the `data` is a `TiledSubNodeData` type. This could avoid type convertion in `if` logic. For example:

```ts
class Base {
}

class TiledSubNodeData extends Base {
   hello();
}

const pBase: Base = new TiledSubNodeData();

if (isTiledSubNodeData(pBase)) { // pBase is Base type, but it's an instance of TiledSubNodeData, so this function return true. And pBase will be treated as a TiledSubNodeData instance.

    pBase.hello(); // No need to convert it to TiledSubNodeData by `(pBase as TiledSubNodeData).hello();`

}
```

Reply:
See: https://matiashernandez.dev/blog/post/what-are-type-predicates-in-typescript

Reply:
Got it, thanks.